### PR TITLE
ci: onboarding module to ibm registery

### PR DESCRIPTION
**Pull request by rajatagarwal-ibm**
SKIP UPGRADE TEST: Changing example name

### Description

onboarding module

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,15 +21,15 @@ provider "ibm" {
...
 }
 
 # KMS standard key
 module "kms_standard_key" {
   source  = "terraform-ibm-modules/kms-key/ibm"
   version = "latest" # Replace "latest" with a release version to lock into a specific release
-  kms_instance_id = ibm_resource_instance.kms_instance.guid
+  kms_instance_id = "XXxxXXxx-xxxx-XXXX-xxxx-XXxxXXxx"
   key_name                = "my-standard-key"
   standard_key            = true
```
Comment (ocofaigh) [inline README.md:34]: Can we fix up the indentation so all the equal signs are matched up (like the linter would do).

Also replace "latest" with "X.X.X"

File: ibm_catalog.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        {
+          "title": "Create a KMS key",
+          "description": "Create a standard or root key in an IBM KMS instance."
+        },
+        {
+          "title": "Create a KMS key policy",
+          "description": "Create a key policy for a standard or root key in an IBM KMS instance."
+        },
+        {
+          "title": "Add key to the KMS key ring",
```
Comment (ocofaigh) [inline ibm_catalog.json:34]: bundle this into the description of the "Create a KMS key" feature above. Doesn't make sense to have as its own feature

File: ibm_catalog.json [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      "offering_docs_url": "https://github.com/terraform-ibm-modules/terraform-ibm-kms-key/blob/main/README.md",
+      "offering_icon_url": "https://raw.githubusercontent.com/terraform-ibm-modules/documentation/main/icons/key_protect_icon.svg",
+      "features": [
+        {
+          "title": "Create a KMS key",
+          "description": "Create a standard or root key in an IBM KMS instance."
```
Comment (ocofaigh) [inline ibm_catalog.json:27]: Create a standard or root key in an existing IBM KMS instance key ring

File: ibm_catalog.json [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+          "working_directory": "examples/basic",
+          "architecture": {
+            "diagrams": [
+              {
+                "diagram": {
+                  "caption": "Basic example",
+                  "url": "https://raw.githubusercontent.com/terraform-ibm-modules/documentation/main/icons/ModuleExampleDiagram.svg",
+                  "type": "image/svg+xml"
+                },
+                "description": "A simple example that shows how to create a standard and a root key an existing IBM Hyper Protect Crypto Service (HPCS) instance."
```
Comment (ocofaigh) [inline ibm_catalog.json:51]: No need to mention HPCS here - the example supports a key from any KMS instance. So update it to what you have in the example readme which is "A simple example that shows how to create a standard and a root key in an existing IBM Key Management Service (KMS) instance."

File: ibm_catalog.json [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+          "working_directory": "examples/complete",
+          "architecture": {
+            "diagrams": [
+              {
+                "diagram": {
+                  "caption": "Complete example",
+                  "url": "https://raw.githubusercontent.com/terraform-ibm-modules/documentation/main/icons/ModuleExampleDiagram.svg",
+                  "type": "image/svg+xml"
+                },
+                "description": "A complete example shows how to provision a Key Protect instance, a root key and a standard key."
```
Comment (ocofaigh) [inline ibm_catalog.json:68]: A complete example **that** shows

File: examples/basic/README.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# Basic example
+
+A simple example that shows how to create a standard and a root key an existing IBM Key Management Service (KMS) instance.
```
Comment (ocofaigh) [inline examples/basic/README.md:3]: missing "in" -> `root key in an existing`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

File: README.md (hunk 1 above)
Comment (rajatagarwal-ibm) [reply README.md:34]: in other TIM modules, we are using "latest" only - although in ibm_catalog.json its X.X.X

File: ibm_catalog.json (hunk 4 above)
Comment (rajatagarwal-ibm) [reply ibm_catalog.json:51]: In catalogValiadtionValues.json.template its pointing to HPCS that's why I mentioned HPCS here.

Comment (ocofaigh) [reply ibm_catalog.json:47]: Yea but thats only for validation - its not exposed in catalog

File: README.md [hunk 7]
```diff
@@ -19,19 +19,19 @@ provider "ibm" {
 
 # KMS root key
 module "kms_root_key" {
-  source  = "terraform-ibm-modules/kms-key/ibm"
-  version = "latest" # Replace "latest" with a release version to lock into a specific release
-  kms_instance_id = ibm_resource_instance.kms_instance.guid
-  key_name                = "my-root-key"
+  source          = "terraform-ibm-modules/kms-key/ibm"
+  version         = "latest" # Replace "latest" with a release version to lock into a specific release
```
Comment (ocofaigh) [inline README.md:23]: replace latest with X.X.X in both places on this line

File: README.md [hunk 8]
```diff
@@ -19,19 +19,19 @@ provider "ibm" {
...
 
 # KMS standard key
 module "kms_standard_key" {
-  source  = "terraform-ibm-modules/kms-key/ibm"
-  version = "latest" # Replace "latest" with a release version to lock into a specific release
-  kms_instance_id = ibm_resource_instance.kms_instance.guid
-  key_name                = "my-standard-key"
-  standard_key            = true
+  source          = "terraform-ibm-modules/kms-key/ibm"
+  version         = "latest" # Replace "latest" with a release version to lock into a specific release
```
Comment (ocofaigh) [inline README.md:31]: replace latest with X.X.X in both places on this line

File: examples/complete/README.md [hunk 9]
```diff
@@ -0,0 +1,9 @@
+# Complete example
+
+A complete example shows how to provision a Key Protect instance, a root key and a standard key.
```
Comment (ocofaigh) [inline examples/complete/README.md:3]: shows -> showing

File: ibm_catalog.json [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+        "target_terraform",
+        "terraform",
+        "module"
+      ],
+      "keywords": [
+        "terraform",
+        "key protect",
+        "hpcs",
+        "keys",
+        "encryption"
```
Comment (ocofaigh) [inline ibm_catalog.json:18]: kms

**Review by ocofaigh — CHANGES_REQUESTED** [review]
final comments

File: ibm_catalog.json [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
...an existing key ring and IBM Key Management Services (KMS) instance. The KMS can be IBM Key Protect or IBM Cloud Hyper Protect Crypto Services(HPCS) instance. You can specify rotation and deletion policies for the keys.\n\n### Usage\n```hcl\nprovider \"ibm\" {\n  ibmcloud_api_key = \"XXXXXXXXXX\"\n  # Must be the same region the KMS instance is in\n  region           = \"us-south\"\n}\n\n# KMS root key\nmodule \"kms_root_key\" {\n  # Replace \"X.X.X\" with a release version to lock into a specific release\n  source.         = \"https://cm.globalcatalog.cloud.ibm.com/api/v1-beta/offering/source?archive=tgz&kind=terraform&name=terraform-ibm-kms-key&version=X.X.X\"\n  kms_instance_id = ibm_resource_instance.kms_instance.guid\n  key_name        = \"my-root-key\"\n}\n\n# KMS standard key\nmodule \"kms_standard_key\" {\n  # Replace \"X.X.X\" with a release version to lock into a specific release\n  source.         = \"https://cm.globalcatalog.cloud.ibm.com/api/v1-beta/offering/source?archive=tgz&kind=terraform&name=terraform-ibm-kms-key&version=X.X.X\"\n  kms_instance_id = ibm_resource_instance.kms_instance.guid\n  key_name        = \"my-standard-key\"\n  standard_key    = true\n}\n```",
```
Comment (ocofaigh) [inline ibm_catalog.json:22]: Can you update `kms_instance_id = ibm_resource_instance.kms_instance.guid` in both module blocks of the usage to use a mock GUID value?

**Review by ocofaigh — CHANGES_REQUESTED** [review]
one last comment

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**